low-level/framework/utils/salt_util.py

```diff
@@ -104,18 +130,23 @@ def get_consul_port(self):
         consul_port = '8500'
         # Read consul port from sspl.sls
         try:
-            SALT_CMD = f"sudo salt-call pillar.get sspl:DATASTORE:consul_port --output=newline_values_only"
-            subout = subprocess.Popen(SALT_CMD, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-            result = subout.stdout.readlines()
-            if result == [] or result == "":
-                logger.warning("Consul Port fetch failed, using 8500 to connect.")
+            if self.pillar_info is not None:
```

Comment:
so many salt commands we are saving, which will optimize for sure

Reply:
Agree